custom_components/localtuya/fan.py

```diff
@@ -98,16 +106,26 @@ def supported_features(self) -> int:
 
     def status_updated(self):
         """Get state of Tuya fan."""
+        mappings = {
+            "1": [SPEED_LOW, 1],
+            "2": [SPEED_MEDIUM, 1],
+            "3": [SPEED_HIGH, 1],
+            "4": [SPEED_HIGH, 1],
+            "low": [SPEED_LOW, 2],
+            "medium": [SPEED_MEDIUM, 2],
+            "high": [SPEED_HIGH, 2],
+            "auto": [SPEED_LOW, 2],
+        }
+
+        dps_id = "%s" % self._dps_id
+
         self._is_on = self._status["dps"]["1"]
```

Comment:
Here is where `self._dps_id` should really go:

    self._is_on = self.dps(self._dps_id)